dm/dm/master/server.go

```diff
@@ -2652,44 +2713,49 @@ func (s *Server) StartValidation(ctx context.Context, req *pb.StartValidationReq
 		return resp2, err2
 	}
 	resp := &pb.StartValidationResponse{}
-	if req.Mode != config.ValidationFull && req.Mode != config.ValidationFast {
-		resp.Result = false
-		resp.Msg = fmt.Sprintf("validation mode should be either `%s` or `%s`", config.ValidationFull, config.ValidationFast)
+
+	subTaskCfgs, msg, ok := s.checkStartValidationParams(req)
+	if !ok {
+		resp.Msg = msg
 		return resp, nil
 	}
-	subTaskCfgs = s.scheduler.GetSubTaskCfgsByTaskAndSource(req.TaskName, req.Sources)
-	if len(subTaskCfgs) == 0 {
-		resp.Result = false
-		name := req.TaskName
-		if name == "" {
-			name = "all"
-		}
-		resp.Msg = fmt.Sprintf("fail to get subtask config by task name `%s` and sources `%v`", name, req.Sources)
-		// nolint:nilerr
-		return resp, nil
+
+	mode := config.ValidationFull
+	if req.Mode != nil {
+		mode = req.GetModeValue()
 	}
-	// TODO: wait for #4479 and update the validator stage in etcd
-	// get the validator stage: common.StageValidatorKeyAdapter.Encode(source, task-name)
-	// if no validator exists: update the subtask config & etcd.Put(validator stage running)
-	// if the validator stage is `RUNNING`: then report error
-	// otherwise: update the subtask config & etcd.Put(validator stage: running)
+	startTime := req.GetStartTimeValue()
+	tasks := make([]string, 0)
+	changedSubtaskCfgs := make([]config.SubTaskConfig, 0)
+	validatorStages := make([]ha.Stage, 0)
 	for taskName := range subTaskCfgs {
 		for sourceID := range subTaskCfgs[taskName] {
 			cfg := subTaskCfgs[taskName][sourceID]
-			cfg.ValidatorCfg.Mode = req.Mode
+			if !s.scheduler.ValidatorEnabled(taskName, sourceID) {
+				cfg.ValidatorCfg.Mode = mode
+				cfg.ValidatorCfg.StartTime = startTime
+				changedSubtaskCfgs = append(changedSubtaskCfgs, cfg)
+				log.L().Info("enable validator",
```

Comment:
in above comment we use "resume" and "start", but here the corresponding words become "start" and "enable"? Better to unify them for other developer's reading

Reply:
unified to `enable`(like create+start in task operation), `start`(like start in task op)